xapian-core/weight/bm25weight.cc

```diff
@@ -48,8 +60,8 @@ BM25Weight::init(double factor)
     Xapian::doccount tf = get_termfreq();
 
     double tw = 0;
-    if (get_rset_size() != 0) {
-	Xapian::doccount reltermfreq = get_reltermfreq();
+    if (get_rset_size() != 0 && !bm25+) {
+       	Xapian::doccount reltermfreq = get_reltermfreq();
```

Comment:
This line has come up as changed because the tab has been changed to spaces. If possible, try to configure your editor to either leave unedited lines as is, or to use tabs where possible. (Tabs followed by spaces is fine.)